src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -672,11 +705,36 @@ std::unique_ptr<Object::DeleteOp> D4NFilterObject::get_delete_op()
 
 int D4NFilterObject::D4NFilterReadOp::prepare(optional_yield y, const DoutPrefixProvider* dpp)
 {
+  ldpp_dout(dpp, 0) << "D4NFilterObject::" << __func__ << "()" << dendl;
+  auto blockDir = source->driver->get_block_dir(); 
+  rgw::d4n::CacheBlock block;
+  block.cacheObj.objName = source->get_name(); // for simple ops, the name will be equivalent to the oid, so we want to use the name to retrieve the block
```

Comment:
This extra logic to get the block and then check whether there is a delete marker is not needed. get_obj_attrs_from_cache() will return an -ENOENT in case of a delete marker, correct?